web-bluetooth/gap-characteristics.js

```diff
@@ -0,0 +1,186 @@
+  log('Requesting Bluetooth Device...');
+  navigator.bluetooth.requestDevice(
+    {filters: [{ name: '' }], optionalServices: ['generic_access']})
+  .then(device => {
+    log('Connecting to GATT Server...');
+    return device.gatt.connect();
+  })
+  .then(server => {
+    log('Getting Generic Access Profile...');
+    return server.getPrimaryService('generic_access');
+  })
+  .then(service => {
+    log('Getting GAP Characteristics...');
+    return service.getCharacteristics();
+  })
+  .then(characteristics => {
+    characteristics.forEach(characteristic => {
+      switch (characteristic.uuid) {
+
+        case BluetoothUUID.getCharacteristic('gap.appearance'):
+          characteristic.readValue().then(value => {
+            log('> Appearance: ' +
+                getDeviceType(value.getUint16(0, true /* Little Endian */)));
+          });
+          break;
+
+        case BluetoothUUID.getCharacteristic('gap.device_name'):
+          characteristic.readValue().then(value => {
+            log('> Device Name: ' + new TextDecoder().decode(value));
+          });
+          break;
+
+        case BluetoothUUID.getCharacteristic('gap.peripheral_preferred_connection_parameters'):
+          characteristic.readValue().then(value => {
+            log('> Peripheral Preferred Connection Parameters: ');
+            log('  >  Minimum Connection Interval: ' +
+                (value.getUint8(0) | value.getUint8(1) << 8) * 1.25 + 'ms');
+            log('  >  Maximum Connection Interval: ' +
+                (value.getUint8(2) | value.getUint8(3) << 8) * 1.25 + 'ms');
+            log('  >  Slave Latency: ' +
+                (value.getUint8(4) | value.getUint8(5) << 8) + 'ms');
+            log('  >  Connection Supervision Timeout Multiplier: ' +
+                (value.getUint8(6) | value.getUint8(7) << 8));
+          });
+          break;
+
+        case BluetoothUUID.getCharacteristic('gap.central_address_resolution_support'):
+          characteristic.readValue().then(value => {
+            let supported = value.getUint8(0);
+            if (supported === 0) {
+              log('> Central Address Resolution: Not Supported');
+            } else if (supported === 1) {
+              log('> Central Address Resolution: Supported');
+            } else {
+              log('> Central Address Resolution: N/A');
+            }
+          });
+          break;
+
+        case BluetoothUUID.getCharacteristic('gap.peripheral_privacy_flag'):
+          characteristic.readValue().then(value => {
+            let flag = value.getUint8(0);
+            if (flag === 1) {
+              log('> Peripheral Privacy Flag: Enabled');
+            } else {
+              log('> Peripheral Privacy Flag: Disabled');
+            }
+          });
+          break;
+
+        default: log('> Unknown Characteristic: ' + characteristic.uuid);
+      }
+    });
+  })
+  .catch(error => {
+    log('Argh! ' + error);
+  });
+}
+
+function getDeviceType(value) {
```

Comment:
What do you think about storing this mapping in an 

```
valueToDeviceType = {
  0: 'Unknown',
  ...
};
```

in the outer scope, and then changing this function to

```
return value + (value in valueToDeviceType ?
                ' (' + valueToDeviceType[value] + ')' :
                '');
```
(If you can assume support for template literals, that would be cleaner, too.)
That would lead to a less intimidating `switch()` and you might theoretically be able to reuse that `valueToDeviceType` mapping elsewhere.